### ISSUE TITLE
[Protocol3] Test failures on master

### ISSUE THREAD

**Issue by dong77** [Loopring/protocols#624]
`npm run test -- transpiled/test/testExchangeBlocks.js` on the master branch has the following test failures:

https://gist.github.com/dong77/c26737656b35fac7be87b4c94471717a

@yueawang  can you run the test to verify if you see the same errors? I think https://github.com/Loopring/protocols/commit/137e598900b84fa78f94798c2ba021881c369791 caused the issue.

**Comment by dong77** [comment]
@yueawang I reverted your commit that added more tests, please redo it and run the tests again (please make sure you delete /blocks directory before you run the test)

**Comment by dong77** [comment]
@yueawang I reverted your commit that added more tests, please redo it and run the tests again (please make sure you delete /blocks directory before you run the test)

**Comment by yueawang** [comment]
I found the reason why lots of cases fail after my case. It because of sync() which does not only sync events of the exchange now running but also of those exchanges created and completed before, i.e. all exchanges forged in UniversalRegistry.
Back to my case, to check if the code path is covered, I created an invalid block to check the path and then emit an event, which blocks sync() happens in the following cases.
To save time, I tried locally with a small set (my case and some failed cases). It can be fixed by sync only the latest exchange, but as it is a fundamental change of the test framework, maybe now we can just leave it alone with my cases reverted.
